runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -679,6 +682,54 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		stringStack[stringStackSize - 1] += titleBuilder.toString();
 	}
 
+	private void setLimitReset(int itemId)
+	{
+		String lastLimit = getLimitReset(itemId);
+		if (!lastLimit.equals("None"))
+		{
+			return; // do not update if previous limit has not expired
+		}
+		String itemIdStr = Integer.toString(itemId);
+		Instant currentDateTime = Instant.now();
+		configManager.setConfiguration("grandexchange",
+			client.getUsername().toLowerCase() + itemIdStr, currentDateTime);
+	}
+
+	private String getLimitReset(int itemId)
+	{
+		String itemIdStr = Integer.toString(itemId);
+		Instant lastDateTime = configManager.getConfiguration("grandexchange",
+			client.getUsername().toLowerCase() + itemIdStr, Instant.class);
+		if (lastDateTime == null)
+		{
+			return "None";
+		}
+		Instant now = Instant.now();
+		long minuteDifference = ChronoUnit.MINUTES.between(lastDateTime, now);
+		if (minuteDifference / 60.0 > 4.0)
+		{
+			return "None";
+		}
+		else
+		{
+			int minsLeft = 240 - (int) minuteDifference;
+			int hours = minsLeft / 60;
+			int minutes = minsLeft % 60;
+			String limitResetString = hours + ":" + minutes;
```

Comment:
Use a date time formatter instead of building this manually. Also this will show eg `1:1` for 1 hour and 1 minute which is wrong.

Reply:
Why is 1 hour 1 minute wrong? 1:1 is correct isnt it

Reply:
Typically clocks show that as 1:01 or 01:01.

Reply:
its not meant to be a clock. it was originally supposed to show "x hours, y minutes" but i decided to change it to "x:y" to shorten the message

Reply:
Clocks, timers, whatever you want to call it.

Reply:
@Adam- wasnt able to figure out how to make a countdown timer using datetimeformatter. All solutions I found were building it manually like how I'm doing it now.

Reply:
Hmm, okay. At the very least 1 hour 1 min should not be `1:1`

Reply:
@Adam- Yeah I agree, I've changed it with a string formatter to 1:01